### PR TITLE
Clarify prevector::erase and avoid swap-to-clear

### PR DESCRIPTION
**Pull request by sipa**

### CODE REVIEW

**Comment by laanwj** [comment]
utACK https://github.com/bitcoin/bitcoin/pull/10534/commits/38a52358f9f145d16f1e4ebab0266740cee86868, thanks for adding the clarifying comment.

File: src/prevector.h [hunk 1]
```diff
@@ -387,6 +387,12 @@ class prevector {
     }
 
     iterator erase(iterator first, iterator last) {
+        // Erase is not allowed the change the object's capacity. That means
```
Comment (sdaftuar) [inline src/prevector.h]: nit: "to" change the object's capacity.

**Review by sdaftuar — APPROVED** [review]
Looks good, thanks for adding.  One typo in the comment.

**Comment by TheBlueMatt** [comment]
utACK 38a52358f9f145d16f1e4ebab0266740cee86868 +/- grammar

**Comment by sipa** [comment]
Fixed.

**Pull request merged**